### PR TITLE
Fix formats not found message not appearing

### PR DESCRIPTION
**Pull request by Karthik99999**
well actually "No formats found" whatever. Maybe it shouldn't rely on `bufs` being empty?

### CODE REVIEW

**Comment by mia-pi-git** [comment]
Probably should check when looping over `bufs` to see if there's any text in the string to begin with.

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -1318,7 +1318,7 @@
 				);
 			}
 			var html = '';
-			if (!bufs.length) {
+			if (!bufs.every(function (buf) { return !!buf; })) {
```
Comment (urkerab) [inline js/client-mainmenu.js]: This logic is wrong, which is why you added the length check to try to fix it.

File: js/client-mainmenu.js [hunk 2]
```diff
@@ -1318,7 +1318,7 @@
 				);
 			}
 			var html = '';
-			if (!bufs.length) {
+			if (!bufs.length || !bufs.every(function (buf) { return !!buf; })) {
```
Comment (urkerab) [inline js/client-mainmenu.js:1321]:
```suggestion
			if (bufs.every(function (buf) { return !buf; })) {
```

File: js/client-mainmenu.js (hunk 1 above)
Comment (Karthik99999) [reply js/client-mainmenu.js]: I still need the length check in case `bufs` is empty

Comment (Karthik99999) [reply js/client-mainmenu.js]: actually nvm I see what you mean now

**Comment by DaWoblefet** [comment]
Thanks all!

**Pull request merged**